src/Model/CustomerTokenService.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/module-customer-graph-ql
+ * @link https://github.com/scandipwa/module-customer-graph-ql
+ */
+
+declare(strict_types=1);
+
+namespace ScandiPWA\CustomerGraphQl\Model;
+
+use Magento\Customer\Api\AccountManagementInterface;
+use Magento\Framework\Exception\EmailNotConfirmedException;
+use Magento\Integration\Model\CredentialsValidator;
+use Magento\Integration\Model\Oauth\TokenFactory as TokenModelFactory;
+use Magento\Integration\Model\ResourceModel\Oauth\Token\CollectionFactory as TokenCollectionFactory;
+use Magento\Integration\Model\Oauth\Token\RequestThrottler;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Integration\Model\CustomerTokenService as SourceCustomerTokenService;
+
+/**
+ * Class CustomerTokenService
+ * @package ScandiPWA\CustomerGraphQl\Model
+ */
+class CustomerTokenService extends SourceCustomerTokenService
+{
+    /**
+     * Token Model
+     *
+     * @var TokenModelFactory
+     */
+    private $tokenModelFactory;
```

Comment:
preferrable to cast private methods as protected instead

Reply:
On another note, not sure what our policy on PHP versions is.
If this will be included only in new PWA, which is supported on only latest Magento versions, which require high enough PHP versions(7.4+) - then you can set direct types on the properties

if we need support for older PHP versions in the core still, nevermind